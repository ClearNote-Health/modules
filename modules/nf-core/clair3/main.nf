process CLAIR3 {
    tag "$meta.id"
    label 'process_high'

    conda 'bioconda::clair3=1.0.0'
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/clair3:1.0.0--py39hb9dc472_1' :
        'quay.io/biocontainers/clair3:1.0.0--py39hb9dc472_1' }"

    input:
    tuple val(meta), path(bam), path(bai)
    tuple val(meta2), path(fasta)
    tuple val(meta3), path(fai)
    val (platform)
    val (model)

    output:
    tuple val(meta), path('${prefix}/*'), emit: vcf
    path "versions.yml"                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    run_clair3.sh \
    --bam_fn=$bam \
    --ref_fn=$fasta \
    --threads=$task.cpus \
    --output="${meta.id}" \
    --platform=${platform} \
    --model_path=${model}
    $args
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        clair3: \$( /usr/local/bin/run_clair3.sh --version | sed 's/Clair3 v//' )
    END_VERSIONS
    """
    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    mkdir ${prefix}
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        clair3: \$( /usr/local/bin/run_clair3.sh --version | sed 's/Clair3 v//' )
    END_VERSIONS
    """
}
