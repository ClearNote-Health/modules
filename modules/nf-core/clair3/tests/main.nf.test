nextflow_process {

    name "Test Process CLAIR3"
    script "../main.nf"
    process "CLAIR3"

    tag "modules"
    tag "modules_nfcore"
    tag "clair3"

    test("sarscov2 - bam") {

        when {
            process {
                """
                input[0] = [ [ id:'test' ], // meta map
                            file(params.modules_testdata_base_path + '/genomics/sarscov2/nanopore/bam/test.sorted.bam', checkIfExists: true),
                            file(params.modules_testdata_base_path + '/genomics/sarscov2/nanopore/bam/test.sorted.bam.bai', checkIfExists: true),
                        ]
                input[1] = [ [ id:'test' ],
                            file(params.modules_testdata_base_path + '/genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                        ]
                input[2] = [ [ id:'test' ],
                            file(params.modules_testdata_base_path + '/genomics/sarscov2/genome/genome.fasta.fai', checkIfExists: true)
                        ]
                input[3] = "ont"
                input[4] = "/usr/local/bin/models/r941_prom_sup_g5014"
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("sarscov2 - bam - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [ [ id:'test' ], // meta map
                            file(params.modules_testdata_base_path + '/genomics/sarscov2/nanopore/bam/test.sorted.bam', checkIfExists: true),
                            file(params.modules_testdata_base_path + '/genomics/sarscov2/nanopore/bam/test.sorted.bam.bai', checkIfExists: true),
                        ]
                input[1] = [ [ id:'test' ],
                            file(params.modules_testdata_base_path + '/genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                        ]
                input[2] = [ [ id:'test' ],
                            file(params.modules_testdata_base_path + '/genomics/sarscov2/genome/genome.fasta.fai', checkIfExists: true)
                        ]
                input[3] = "ont"
                input[4] = "/usr/local/bin/models/r941_prom_sup_g5014"
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
