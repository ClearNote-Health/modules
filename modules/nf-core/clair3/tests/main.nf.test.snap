{
    "sarscov2 - bam - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "vcf": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-06-16T05:01:02.37044703"
    },
    "sarscov2 - bam": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "vcf": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-06-16T05:00:52.108250388"
    }
}